examples/tiled_ipywidget.ipynb

```diff
@@ -0,0 +1,263 @@
+    "def get_tiled_uri_metadata_field_entries(base_uri, uri_additional, metadata_field):\n",
+    "    \"\"\"\n",
+    "    base_uri - str, tiled base uri (ex: looks like http://127.0.0.1:8888)\n",
+    "    uri_additional - str, the add on path to the base_uri (ex: could be something like \"node_name_1\" or \"\"\n",
+    "                    based on your file structure)\n",
+    "    metadata_field - str, the field to extract\n",
+    "    \"\"\"\n",
+    "    try:\n",
+    "        full_tiled_uri = base_uri + \"/api/v1/metadata/\" + uri_additional\n",
+    "        client = from_uri(full_tiled_uri)\n",
+    "        dropdown_list = []\n",
+    "\n",
+    "        for experiment in client:\n",
+    "            if \"seg_result\" in client[experiment]:\n",
+    "                child_node_client = client[experiment][\"seg_result\"]\n",
+    "                scan_uri = child_node_client.uri\n",
+    "                metadata = child_node_client.metadata\n",
+    "                metadata_field_entry = metadata[metadata_field]\n",
+    "                dropdown_list.append((metadata_field_entry, scan_uri))\n",
+    "\n",
+    "        return dropdown_list\n",
+    "\n",
+    "    except Exception as e:\n",
+    "        error_message = (\n",
+    "            f\"Error retrieving experiment names: {e} due to {traceback.format_exc()}\"\n",
+    "        )\n",
+    "        print(error_message)\n",
+    "        return []"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def tiled_widget_refresh_callback(dropdown_widget):\n",
+    "    # should be the same string as the next cell, because the refresh fetches the same tiled uri\n",
+    "    new_list_experiments = get_tiled_uri_metadata_field_entries(\n",
+    "        tiled_base_uri, additional_string_tiled_uri, metadata_filed\n",
+    "    )\n",
+    "    dropdown_widget.options = new_list_experiments\n",
+    "    dropdown_widget.value = None"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scan_uri_metadata_tuples = get_tiled_uri_metadata_field_entries(\n",
+    "    tiled_base_uri, additional_string_tiled_uri, metadata_filed\n",
+    ")\n",
+    "\n",
+    "tiled_file_widget = widgets.Dropdown(\n",
+    "    options=scan_uri_metadata_tuples,\n",
+    "    value=None,\n",
+    "    description=\"Experiment name:\",\n",
+    "    style={\"description_width\": \"initial\"},\n",
+    ")\n",
+    "\n",
+    "tiled_widget_refresh_button = widgets.Button(description=\"Refresh\")\n",
+    "\n",
+    "widget_box = widgets.HBox([tiled_file_widget, tiled_widget_refresh_button])\n",
+    "\n",
+    "display(widget_box)\n",
+    "\n",
+    "\n",
+    "def tiled_refresh_button_click(button):\n",
+    "    tiled_widget_refresh_callback(tiled_file_widget)\n",
```

Comment:
I have not worked with IPyWidgets before - is wrapping the function like this needed for the callback?